src/materialized/tests/util.rs

```diff
@@ -34,7 +34,7 @@ pub struct Config {
 impl Default for Config {
     fn default() -> Config {
         Config {
-            data_directory: None,
+            data_directory: tempfile::tempdir().unwrap().into_path(),
```

Comment:
Unwrapping here is a little subpar. I suggest deleting the `data_directory` member from `Config` entirely, and deferring the creation of the temporary directory until `start_server`, where you can return an error.

Reply:
Also, calling `into_path` means the temporary directory is no longer going to be cleaned up, which means they'll start to pile up if you run the tests a lot. I think you'll want to do something like

```rust
pub fn start_server(config: Config) -> Result<(Server, postgres::Client), Box<dyn Error>> {
    let mut runtime = Runtime::new()?;
    let temp_dir = tempfile::tempdir()?;
    let inner = runtime.block_on(materialized::serve(materialized::Config {
        // ...
        data_directory: temp_dir.path(),
    }))?;
    let server = Server {
        inner,
        _runtime: runtime,
        _temp_dir: temp_dir,
    };
    // ...
    Ok((server, ...))
}
```

so that the `temp_dir` outlives the server and then is automatically cleaned up when the server is dropped.

Reply:
Oh, just saw that you mentioned that question in the PR description! I did have an approach to recommend! :D